public/containers/FlushIndexModal/FlushIndexModal.tsx

```diff
@@ -0,0 +1,182 @@
+    }
+    const indexPayload = unBlockedItems.join(",");
+    const result = await services.commonService.apiCaller({
+      endpoint: "indices.flush",
+      data: {
+        index: indexPayload,
+      },
+    });
+    if (result && result.ok) {
+      const flushedItems = flushAll ? "all open indexes" : `[${indexPayload}]`;
+      coreServices.notifications.toasts.addSuccess(`Flush ${flushedItems} successfully`);
+    } else {
+      coreServices.notifications.toasts.addDanger(result.error);
+    }
+    onClose();
+  }, [unBlockedItems, services, coreServices, onClose, flushAll]);
+
+  useEffect(() => {
+    if (!!services && visible) {
+      switch (flushTarget) {
+        case "alias":
+          filterBlockedItems<IAlias>(services, selectedItems as IAlias[], IndexOpBlocksType.Closed, aliasBlockedPredicate)
+            .then((filterBlockedItems) => {
+              if (visible) {
+                setBlockedItems(filterBlockedItems.blockedItems.map((item) => item.alias));
+                setUnBlockedItems(filterBlockedItems.unBlockedItems.map((item) => item.alias));
+              }
+            })
+            .catch((err) => {
+              if (visible) {
+                setUnBlockedItems((selectedItems as IAlias[]).map((item) => item.alias));
+              }
+            });
+          break;
+        case "data stream":
+          filterBlockedItems<DataStream>(services, selectedItems as DataStream[], IndexOpBlocksType.Closed, dataStreamBlockedPredicate)
+            .then((filterBlockedItems) => {
+              if (visible) {
+                setBlockedItems(filterBlockedItems.blockedItems.map((item) => item.name));
+                setUnBlockedItems(filterBlockedItems.unBlockedItems.map((item) => item.name));
+              }
+            })
+            .catch((err) => {
+              if (visible) {
+                setUnBlockedItems((selectedItems as DataStream[]).map((item) => item.name));
+              }
+            });
+          break;
+        default:
+          filterBlockedItems<CatIndex>(services, selectedItems as CatIndex[], IndexOpBlocksType.Closed, indexBlockedPredicate)
+            .then((filterBlockedItems) => {
+              if (visible) {
+                setBlockedItems(filterBlockedItems.blockedItems.map((item) => item.index));
+                setUnBlockedItems(filterBlockedItems.unBlockedItems.map((item) => item.index));
+              }
+            })
+            .catch((err) => {
+              if (visible) {
+                setUnBlockedItems((selectedItems as CatIndex[]).map((item) => item.index));
+              }
+            });
+      }
+    } else {
+      setBlockedItems([]);
+      setUnBlockedItems([]);
+    }
+  }, [visible, flushTarget, selectedItems]);
+
+  if (!visible) {
+    return null;
+  }
+
+  return (
+    <EuiModal onClose={onClose}>
+      <EuiModalHeader>
+        <EuiModalHeaderTitle data-test-subj="Flush Modal Title">Flush {flushTarget}</EuiModalHeaderTitle>
```

Comment:
Recommend using camel case for data-test-subj: https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/DEVELOPER_GUIDE.md#camel-case-id-and-data-test-subj